pkg/kapp/app/app_filter.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import "time"
+
+type AppFilter struct {
+	CreatedAtBeforeTime *time.Time
+	CreatedAtAfterTime  *time.Time
+
+	Labels []string
+}
+
+func (f AppFilter) Apply(apps []App) []App {
+	var result []App
+
+	for _, app := range apps {
+		if f.Matches(app) {
+			result = append(result, app)
+		}
+	}
+	return result
+}
+
+func (f AppFilter) Matches(app App) bool {
+	if f.CreatedAtBeforeTime != nil {
+		if lc, _ := app.LastChange(); lc.Meta().StartedAt.After(*f.CreatedAtBeforeTime) {
```

Comment:
Although inline check seems fine but just wondering if we should call `LastChange()` before the if condition and do an error check as well.

Reply:
Yeah added error check.